okhttp-dnsoverhttps/src/main/kotlin/okhttp3/dnsoverhttps/DnsOverHttps.kt

```diff
@@ -67,7 +69,31 @@ class DnsOverHttps internal constructor(
       }
     }
 
-    return lookupHttps(hostname)
+    return try {
+      lookupHttps(hostname)
+    } catch (bestFailure: UnknownHostException) {
+      if (checkIsIpAddress(hostname)) {
+        try {
+          InetAddress.getAllByName(hostname).toList()
+        } catch (e: UnknownHostException) {
+          throw bestFailure
+        }
+      } else {
+        throw bestFailure
+      }
+    }
+  }
+
+  private fun checkIsIpAddress(hostname: String): Boolean {
```

Comment:
Should use our existing function, `canParseAsIpAddress()` from `okhttp3.internal`. We definitely don’t want to functions to convert IP addresses to strings.

Reply:
Sure, I missed that there is a existing function.